### PR TITLE
add AWS autoscaler installer

### PR DESCRIPTION
**Pull request by metadave**
- default cluster autoscaler max is 4 * KOPS_NODE_COUNT
- increases default Deis router HPA to 4 * KOPS_NODE_COUNT
- needs to be squashed

### CODE REVIEW

**Comment by jgmize** [comment]
As we discussed earlier, we also need to adjust max_size in kops' kubernetes.tf output.

**Comment by metadave** [comment]
I pushed a commit to set the `max_size` to 4 * KOPS_NODE_COUNT

File: k8s/install/stage2_functions.sh [hunk 1]
```diff
@@ -260,14 +260,39 @@ config_deis_router_hpa() {
...
+        jq ".awsRegion=\"${KOPS_REGION}\"" | j2y > "${OUTPUT_FILE}"
+
+    kubectl create namespace "${AUTOSCALER_NAMESPACE}"
+    helm install stable/aws-cluster-autoscaler \
+        --name aws-cluster-autoscaler \
+        --namespace "${AUTOSCALER_NAMESPACE}" \
+        -f "${OUTPUT_FILE}"
+
+    # set ASG max size so to allow the cluster autoscaler to scale up
+    sed -i "s/max_size = $KOPS_NODE_COUNT/max_size = $DEFAULT_MAX/" ./out/terraform/kubernetes.tf
```
Comment (jgmize) [inline k8s/install/stage2_functions.sh]: This implies an additional `terraform apply` step, should we document that?

Comment (metadave) [inline k8s/install/stage2_functions.sh]: no, it should be in the stage1 and run with all the other terraform. fixed.

Comment (jgmize) [inline k8s/install/stage2_functions.sh]: even better, thanks :)

**Pull request merged**